core/src/main/scala/kafka/metrics/clientmetrics/ClientMetricsConfig.scala

```diff
@@ -0,0 +1,123 @@
+package kafka.metrics.clientmetrics
+
+import kafka.metrics.clientmetrics.ClientMetricsCacheOperation.{CM_SUBSCRIPTION_ADDED, CM_SUBSCRIPTION_DELETED, CM_SUBSCRIPTION_UPDATED}
+import kafka.metrics.clientmetrics.ClientMetricsConfig.ClientMetrics.{AllMetricsFlag, ClientMatchPattern, DeleteSubscription, PushIntervalMs, SubscriptionMetrics, configDef}
+import org.apache.kafka.common.config.ConfigDef
+import org.apache.kafka.common.config.ConfigDef.Importance.MEDIUM
+import org.apache.kafka.common.config.ConfigDef.Type.{BOOLEAN, INT, LIST}
+import org.apache.kafka.common.errors.InvalidRequestException
+
+import java.util
+import java.util.Properties
+import java.util.concurrent.ConcurrentHashMap
+import scala.collection.mutable.ListBuffer
+
+/**
+ * Client metric configuration related parameters and the supporting methods like validation and update methods
+ * are defined in this class.
+ * For more information please look at kip-714:
+ * https://cwiki.apache.org/confluence/display/KAFKA/KIP-714%3A+Client+metrics+and+observability
+ */
+object ClientMetricsConfig {
+
+  class SubscriptionGroup(subscriptionGroup: String,
+                          subscribedMetrics: List[String],
+                          var matchingPatternsList: List[String],
+                          pushIntervalMs: Int,
+                          allMetricsSubscribed: Boolean = false) {
+    def getId = subscriptionGroup
+    def getPushIntervalMs = pushIntervalMs
+    val clientMatchingPatterns = CmClientInformation.parseMatchingPatterns(matchingPatternsList)
+    def getClientMatchingPatterns = clientMatchingPatterns
+    def getSubscribedMetrics = subscribedMetrics
+    def getAllMetricsSubscribed = allMetricsSubscribed
+  }
+
+  object ClientMetrics {
+    // Properties
+    val SubscriptionMetrics = "client.metrics.subscription.metrics"
+    val ClientMatchPattern = "client.metrics.subscription.client.match"
+    val PushIntervalMs = "client.metrics.push.interval.ms"
+    val AllMetricsFlag = "client.metrics.all"
+    val DeleteSubscription = "client.metrics.delete.subscription"
+
+    val DEFAULT_PUSH_INTERVAL = 5 * 60 * 1000 // 5 minutes
+
+    // Definitions
+    val configDef = new ConfigDef()
+      .define(SubscriptionMetrics, LIST, MEDIUM, "List of the subscribed metrics")
+      .define(ClientMatchPattern, LIST, MEDIUM, "Pattern used to find the matching clients")
+      .define(PushIntervalMs, INT, DEFAULT_PUSH_INTERVAL, MEDIUM, "Interval that a client can push the metrics")
+      .define(AllMetricsFlag, BOOLEAN, false, MEDIUM, "If set to true all the metrics are included")
```

Comment:
I am thinking that this a flag set by the AdminClient whenever --all-metrics parameter is used. Of course it can be done by AdminClient too. Let me know, I can delete it from here.